core/deployment/src/main/java/io/quarkus/runner/RuntimeClassLoader.java

```diff
@@ -90,24 +90,9 @@ public RuntimeClassLoader(ClassLoader parent, Path applicationClasses, Path fram
         }
 
         // TODO: some superugly hack for bean provider
-        byte[] data = resources.get(name);
-        if (data != null) {
-            URL url = new URL(null, "quarkus:" + name + "/", new URLStreamHandler() {
-                @Override
-                protected URLConnection openConnection(final URL u) throws IOException {
-                    return new URLConnection(u) {
-                        @Override
-                        public void connect() throws IOException {
-                        }
-
-                        @Override
-                        public InputStream getInputStream() throws IOException {
-                            return new ByteArrayInputStream(resources.get(name));
-                        }
-                    };
-                }
-            });
-            return Collections.enumeration(Collections.singleton(url));
+        URL resource = getQuarkusResource(name);
+        if (resource != null) {
+            return Collections.enumeration(Collections.singleton(resource));
```

Comment:
This one should aggregate the resource with the result of `findApplicationResource(name)` as well as the result of `super.getResources(name)`.  So you'll probably need either a custom enumeration type, or to eagerly fetch all resources (including `super.getResources()`) into a collection and enumerate over that.